cpu/esp_common/esp-now/esp_now_netdev.c

```diff
@@ -53,21 +61,48 @@
 #define ESP_NOW_AP_PREFIX        "RIOT_ESP_"
 #define ESP_NOW_AP_PREFIX_LEN    (strlen(ESP_NOW_AP_PREFIX))
 
+#ifdef MCU_ESP32
+
+#define esp_const_uint8_t        const uint8_t
+#define esp_now_set_self_role(r)
+
+#else
+
+#define ESP_NOW_ROLE_IDLE        (0)
+#define ESP_NOW_ROLE_CONTROLLER  (1)
+#define ESP_NOW_ROLE_SLAVE       (2)
+#define ESP_NOW_ROLE_COMBO       (3)
+#define ESP_NOW_KEY_LEN          (16)
+
+#define ESP_OK                   (0)
+
+#ifndef IRAM_ATTR
+#define IRAM_ATTR                IRAM
+#endif
+
+#define esp_const_uint8_t        uint8_t
```

Comment:
Let's move [this discussion](https://github.com/RIOT-OS/RIOT/commit/f412c9101d5158014f88f58559abe710f59da64c#diff-24495fdc92b602d8c37d024123b775a5R83) back to the PR ;-)

Reply:
I'd prefer not to hide the non-constness. If it is required to be not const for certain implementations, then remove the const qualifier completely.

Reply:
Then, the only way is to embed functions with different signatures in `#ifdef MCU_ESP32 .. #else ... #endif`

Reply:
> > Not const :-/
> 
> Yes, that's the trick to be able to use as much as possible shared code and not to need to embed all function signatures in
> 
> ```c
> #ifdef MCU_ESP32
> static IRAM_ATTR void esp_now_recv_cb(const uint8_t *mac, const uint8_t *data, int len)
> #else
> static IRAM_ATTR void esp_now_recv_cb(uint8_t *mac, uint8_t *data, uint8_t len)
> #endif
> ```
> 
> Unfortunately, even the ESP-NOW API function seems to be the same, the parameters of ESP8266 version are always non-const while the parameters of ESP32 are const 
> 
> This is especially a problem when a certain function signature is required when they are registered as callbacks.

> Then, the only way is to embed functions with different signatures in `#ifdef MCU_ESP32 .. #else ... #endif`

I'd prefer this over intransparent type magic. Alternative: move these functions to CPU-specific implementation?

Reply:
The problem is only the signature, implementations are at least partially the the same.

Reply:
Changed.

Reply:
I'll have another look in a moment